apps/prairielearn/src/lib/file-store.ts

```diff
@@ -176,23 +210,23 @@ export async function getFile(file_id, data_type = 'buffer') {
 
     if (data_type === 'buffer') {
       debug(
-        `get(): s3 fetch file ${result.rows[0].storage_filename} from ${config.fileStoreS3Bucket} and return object with contents buffer and file object`,
+        `get(): s3 fetch file ${result.storage_filename} from ${config.fileStoreS3Bucket} and return object with contents buffer and file object`,
       );
-      buffer = await getFromS3(config.fileStoreS3Bucket, result.rows[0].storage_filename);
+      buffer = await getFromS3(config.fileStoreS3Bucket, result.storage_filename);
     } else {
       debug(
-        `get(): s3 fetch stream ${result.rows[0].storage_filename} from ${config.fileStoreS3Bucket} and return object with contents stream and file object`,
-      );
-      readStream = await getFromS3(
-        config.fileStoreS3Bucket,
-        result.rows[0].storage_filename,
-        false,
+        `get(): s3 fetch stream ${result.storage_filename} from ${config.fileStoreS3Bucket} and return object with contents stream and file object`,
       );
+      readStream = await getFromS3(config.fileStoreS3Bucket, result.storage_filename, false);
     }
   }
 
+  const contents = buffer ?? readStream;
+  if (contents === undefined) {
+    throw new Error(`No contents for file_id ${file_id}`);
+  }
```

Comment:
Nit: you could skip this (technically unnecessary) error handling with some early returns: instead of assigning to `buffer`/`readStream` above, just return `{ file: results, ... }` immediately.